src/System.ServiceModel.Syndication/src/System/ServiceModel/Syndication/SyndicationFeed.cs

```diff
@@ -265,6 +269,44 @@ public TextSyndicationContent Title
             set { _title = value; }
         }
 
+        internal SyndicationLink GetDocumentation()
+        {
+            return _documentation;
+        }
+
+        public SyndicationLink Documentation
+        {
+            get
+            {
+                if (_documentation == null)
+                {
+                    _documentation = TryReadDocumentationFromExtension(ElementExtensions);
+                }
+
+                return _documentation;
+            }
+            set
+            {
+                _documentation = value;
+            }
+        }
+
+        private SyndicationLink TryReadDocumentationFromExtension(SyndicationElementExtensionCollection elementExtensions)
+        {
+            SyndicationElementExtension documentationElement = elementExtensions
+                                      .Where((e) => e.OuterName == Rss20Constants.DocumentationTag && e.OuterNamespace == Rss20Constants.Rss20Namespace)
+                                      .FirstOrDefault();
+
+            if (documentationElement == null)
+                return null;
+
+            using (XmlReader reader = documentationElement.GetReader())
+            {
+                SyndicationLink documentation = new Rss20FeedFormatter().ReadAlternateLink(reader, BaseUri);
+                return documentation;
```

Comment:
ReadAlternateLink uses a UriParser which is why you need to instantiate an instance of Rss20FeedFormatter. This is just going to use the default Uri parser so we lose the benefit of the extensibility here. We should make ReadAlternateLink be a static method which takes a Uri parser as a parameter . Then do one of two things. 1) Have the class which instantiates the SyndicationFeed class set the parser in SyndicationFeed so any custom parser can be passed to ReadAlternateLink. 2) Pass the default Uri parser to ReadAlternateLink as that's what's being used here anyway.

Reply:
Sounds good. I went with option 2).